enhancements/sig-architecture/8-addon-framework/README.md

```diff
@@ -0,0 +1,200 @@
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ClusterManagementAddon
+metadata:
+ name: submariner
+spec:
+ addonMeta:
+   displayName: submariner
+   description: submariner-addon
+ addonConfiguration:
+   crd: submarineraddonconfig.addon.open-cluster-management.io
+   name: submariner-config
+```
+
+#### Enable addon agent on a managed cluster
+
+1. A user creates a ManagedClusterAddon resource of which the name is the same as the name of ClusterManagerAddon resource in the cluster namespace.
+```yaml
+ apiVersion: addon.open-cluster-management.io/v1alpha1
+ kind: ManagedClusterAddon
+ metadata:
+   name: submariner
+   namespace: cluster1
+```
+2. Addon controller on hub deploys the agent by creating manifestworks in cluster namespace.
+3. The agent is started on a managed cluster, and registers back using credential.
+
+#### Communication pattern
+
+An addon agent in a managed cluster will communicate with the hub:
+- The addon agent talks to the kube apiserver of the hub.  For reading desired state and for pushing small quantities of data that are stored in kubernetes API objects, this is the recommended path.
+- The addon agent talks to another endpoint on the hub.  For writing large amounts of data to the hub or sending data that doesn’t fit cleanly into a kubernetes style API, this is the recommended approach.
+
+In both ways, the communication should be properly authenticated/authorized.
+
+##### mTLS for authentication
+
+The addon agent should be able to register to the hub by issuing a csr request and choose the signer. If the addon agent talks to the kube-apiserver of the hub, the agent should use the `kubernetes.io/kube-apiserver-client` to create the csr. Otherwise, the addon agent should use a customized signer for different endpoints. The endpoint, which supposedly belongs to the hub component of the addon should be able to accept the cert signed by the customized signer for this addon.
+
+##### Permission for the addon agent
+
+The addon should have a controller on hub to correctly set the permission for an addon agent to talk to the hub. 
+1. An addon agent may only be able to access certain resources on the cluster namespace on the hub.
+2. An addon agent may only be able to access certain resources on a specified namespace (not cluster namespace) on the hub.
+3. An addon agent may be able to access a certain url on a non kube-apiserver endpoint.
+
+The addon controller on the hub should have a reconcile loop that when an addon is enabled on a cluster, the permission of the addon agent can be set correctly. 
+
+### Addon framework interfaces
+
+To ease the development of addon controllers, it is possible that we provide a framework library to encapsulate the common part, so that developers of addon controllers only need to care about the manifests of addon agents that should be deployed on a managed cluster. 
```

Comment:
Go ahead and enumerate some of the examples here.  As I recall, the credentials themselves took a bit of a dance.

Reply:
An example is added.